src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,74 @@
+package duke.task;
+
+/**
+ * Task is an abstract base class for all types of tasks
+ * that users wish to add into their Bobi task list.
+ * A Task Object encapsulates the task name and status for users to track.
+ *
+ * @author ruo-x
+ */
+public abstract class Task {
+    /** Name of task */
+    private final String taskName;
+
+    /** Status of task */
+    private boolean isDone;
+
+    /**
+     * Constructor of a Task object.
+     *
+     * @param status Status of the task.
+     * @param taskName Name of the task.
+     */
+    public Task(Boolean status, String taskName) {
+        this.taskName = taskName;
+        this.isDone = status;
+    }
+
+    /**
+     * Returns status of task.
+     */
+    public boolean getStatus() {
+        return this.isDone;
+    }
+
+    /**
+     * Returns name of task.
+     */
+    public String getName() {
+        return this.taskName;
+    }
+
+    /**
+     * Marks a task as done.
+     * Status of the task object is changed to true.
+     */
+    public void mark() {
```

Comment:
Could use the `set` here instead for both mark and unmark for it to be even more self explanatory!